selfservice/strategy/oidc/provider_netid.go

```diff
@@ -0,0 +1,100 @@
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+	"path"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/herodot"
+	"github.com/ory/x/httpx"
+)
+
+const (
+	defaultBrokerEndpoint = "https://broker.netid.de"
+)
+
+type ProviderNetID struct {
+	*ProviderGenericOIDC
+}
+
+func NewProviderNetID(
+	config *Configuration,
+	reg dependencies,
+) *ProviderNetID {
+	return &ProviderNetID{
+		ProviderGenericOIDC: &ProviderGenericOIDC{
+			config: config,
+			reg:    reg,
+		},
+	}
+}
+
+func (n *ProviderNetID) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return n.oAuth2(ctx)
+}
+
+func (n *ProviderNetID) oAuth2(ctx context.Context) (*oauth2.Config, error) {
+	endpoint, err := n.endpoint()
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	authUrl := *endpoint
+	tokenUrl := *endpoint
+
+	authUrl.Path = path.Join(authUrl.Path, "/authorize")
```

Comment:
You can also use `ory/x/urlx.CopyWithPath` (I think or maybe it's `AppendPath`) - just make sure it returns a copy of the struct!

Reply:
ok, thanks for the hint :)